### PR TITLE
Move to Jest & upgrades to require Node v5

### PR DESCRIPTION
**Pull request by amir20**
### Proposed changes in this pull request

This a big pull request where it targets only Node5+. 

* Moved to eslint by facebook
* Started using Jest instead (great so far)
* Checked in `yarn.lock` but no really story around using it with travis
* Target Node5 only. Bumps to v3 because it is not backward compatible

#### Checklist
* [x] New tests have been added
* [x] `npm test` passes successfully
* [ ] Documentation has been updated

@amir20 to review

### CODE REVIEW

**Comment by amir20** [comment]
I'll leave this here in case anybody wants to review it.

**Comment by amir20** [comment]
This would force people to update Node. So I would be curious to hear if anybody thinks that's a bad idea. It does clean up the output a lot though.

File: package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "name": "phantom",
   "description": "PhantomJS integration module for NodeJS",
   "homepage": "https://github.com/amir20/phantomjs-node",
-  "version": "2.1.21",
+  "version": "3.0.0",
```
Comment (amir20) [inline package.json]: version bump here

File: package.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
   ],
   "main": "lib/index.js",
   "engines": {
-    "node": ">=0.12"
+    "node": ">=5.0.0"
```
Comment (amir20) [inline package.json]: No longer support < v5

File: .babelrc [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
   "presets": [
-    "es2015"
+    "node5"
```
Comment (amir20) [inline .babelrc]: Now only target Node5.

File: package.json [hunk 4]
```diff
@@ -30,20 +30,32 @@
...
-    "jasmine-co": "^1.1.0",
-    "proxyquire": "^1.7.4"
+    "jest": "^16.0.1"
   },
   "license": "ISC",
   "scripts": {
     "test": "gulp lint test",
     "prepublish": "gulp build"
+  },
+  "jest": {
```
Comment (vampolo) [inline package.json]: by default [jest test environment](https://facebook.github.io/jest/docs/configuration.html#testenvironment-string) is set to `jsdom`. If you are ready to give up the window/globals you can set it to `node` and have faster tests.

File: src/shim/.babelrc [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "presets": [
+    "es2015"
```
Comment (vampolo) [inline src/shim/.babelrc]: this babelrc is different from the top level project one. dunno if it is an oversight or it is intended....

File: package.json (hunk 4 above)
Comment (amir20) [inline package.json]: I had forgotten about that! Thanks. Although, after making that change, I didn't see any improvements.

File: src/shim/.babelrc (hunk 5 above)
Comment (amir20) [inline src/shim/.babelrc]: Yes, this is intended because the `shim` directory is executed in phantomjs which is not node5 compliant.

**Pull request merged**